stl/inc/compare

```diff
@@ -177,30 +147,28 @@ public:
     }
 
     _NODISCARD friend constexpr bool operator<(_Literal_zero, const weak_ordering _Val) noexcept {
-        return 0 < _Val._Value;
+        return _Val > 0;
     }
 
     _NODISCARD friend constexpr bool operator>(_Literal_zero, const weak_ordering _Val) noexcept {
-        return 0 > _Val._Value;
+        return _Val < 0;
     }
 
     _NODISCARD friend constexpr bool operator<=(_Literal_zero, const weak_ordering _Val) noexcept {
-        return 0 <= _Val._Value;
+        return _Val >= 0;
     }
 
     _NODISCARD friend constexpr bool operator>=(_Literal_zero, const weak_ordering _Val) noexcept {
-        return 0 >= _Val._Value;
+        return _Val <= 0;
     }
 
-#ifdef __cpp_impl_three_way_comparison
     _NODISCARD friend constexpr weak_ordering operator<=>(const weak_ordering _Val, _Literal_zero) noexcept {
         return _Val;
     }
 
     _NODISCARD friend constexpr weak_ordering operator<=>(_Literal_zero, const weak_ordering _Val) noexcept {
         return weak_ordering{static_cast<_Compare_ord>(-_Val._Value)};
     }
-#endif // __cpp_impl_three_way_comparison
 
 private:
     _Compare_t _Value;

```

Comment:
I convinced myself long ago that these types would always be inlined away and wouldn't appear on "ABI boundaries", but I'm less happy with that decision today. I'm also concerned that it will very likely be too late to fix the ABI when we discover for certain that it's a problem. So yes, I think I'd like to make this change but I think it should be in a followup PR to keep this uncontroversial bugfix separate from that potentially controversial performance improvement. Is that ok?

Reply:
(Implemented at https://github.com/CaseyCarter/STL/compare/CaseyCarter:compare...CaseyCarter:compare_aggregate.)

Reply:
I'm okay with a separate PR and keeping this cleanup + bugfix only.

Reply:
Submitted #1065.